db/columns/operations/select.py

```diff
@@ -95,14 +95,17 @@ def get_columns_name_from_tables(table_oids, engine, connection_to_use=None, fet
     return column_names
 
 
-def get_columns_name_from_attnums(table_oid, attnums, engine, connection_to_use=None):
+def get_columns_name_from_attnums(table_oid, attnums, engine, connection_to_use=None, fetch_as_map=False):
```

Comment:
I think that changing a function's behaviour based on a boolean switch has a negative effect on refactoring later. It also leads to bigger, more complicated methods. Could you split this behaviour into its own method? Something like `get_column_name_from_attnums_as_map`? Thanks!

Reply:
I get your concern but I think We can break the function when the function gets more complicated. We already have too many of these conversion functions atm.

Reply:
> We can break the function when it gets more complicated

That's a bad idea, because this also affects refactoring. Once you want to break it up, you'll have to walk through its usages (it's a public function) and split them up into whether they're using this version of the function or that version.

So it's not only about the function being complicated, but also about what it does to refactoring. Plus, when you're reading the usage of this method, it's another cognitive step to check what state the boolean switch is in.

> We already have too many of these conversion functions already.

You're not reducing the number of branches with similar-ish code. Due to above reasons, it's better to have functions than switches.

By the way, if you want to get some code reuse, wrap the common code into its own private function:

```
    statement = _get_columns_name_from_attnums([table_oid], attnums, engine, connection_to_use=None)
    column_names_tuple = execute_statement(engine, statement, connection_to_use).fetchall()
```

Maybe `_get_column_names_tuple(table_oid, attnums)`. That will go a longer way than a switch.